src/components/atoms/Icon/Icon.tsx

```diff
@@ -1,11 +1,34 @@
 import React from 'react';
+import styled from 'styled-components';
 import { FontAwesomeIcon, FontAwesomeIconProps } from '@fortawesome/react-fontawesome';
 import { IconDefinition } from '@fortawesome/fontawesome-common-types';
 
+import { Colors } from '../../../constants/colors';
+import { spacing } from '../../../constants/spacing';
+
 interface IconProps extends Omit<FontAwesomeIconProps, 'icon' | 'border'> {
   variant: IconDefinition;
+  bgColor?: Colors[keyof Colors];
 }
 
-export default function Icon({ variant, ...rest }: IconProps) {
-  return <FontAwesomeIcon {...rest} icon={variant} />;
+export const RoundBadge = styled.div<{ color: Colors[keyof Colors] }>`
+  background-color: ${({ color }) => color};
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  border-radius: 50%;
+  width: fit-content;
+  padding: ${spacing[6]};
+`;
+
+export default function Icon({ variant, bgColor, className, ...rest }: IconProps) {
```

Comment:
Hmmm is it ok that the same className prop is passed to the both the badge and the icon - there may be situations where you want to pass different classes to the badge or the icon.

Reply:
If a bgColor has been set the className will be added to only the badge and if not set will be added to the icon. Reason for this was for our tailwind spacing classes and ensuring the margin was set to badge rather than the icon.

I didn't want to add an additional prop to be able to set a className to both, but happy to if you feel there would be a need to

Reply:
Fair, lets only add it if we really need the functionality